new-log-viewer/src/services/LogExportManager.ts

```diff
@@ -0,0 +1,63 @@
+import {downloadBlob} from "../utils/file";
+
+
+/**
+ * Manager for exporting logs to a file.
+ */
+class LogExportManager {
+    /**
+     * Internal buffer which stores decoded chunks of log data.
+     */
+    readonly #chunks: string[] = [];
+
+    /**
+     * Total number of chunks to export.
+     */
+    readonly #numChunks: number;
+
+    /**
+     * Name of the file to export to.
+     */
+    readonly #exportedFileName: string;
+
+    constructor (numChunks: number, fileName: string) {
+        this.#numChunks = numChunks;
+        this.#exportedFileName = fileName;
```

Comment:
```suggestion
        this.#exportedFileName = `${this.#exportedFileName}-exported-${new Date().toISOString().replace(/[:.]/g, "-")}.log`;
```

Reply:
How about `exported-${this.#exportedFileName}-${new Date().toISOString().replace(/[:.]/g, "-")}.log`;

Reply:
I would prefer the suggested one, since users might search the exported logs according to their original file names. In this case, if the start of the file is "exported", it might cause some difficulties for them

Reply:
a counter argument would be: if the user want to find all exported logs, the "exported-" prefix can quickly help them identify the exported files. They can start their file name search with the prefix anyways.

Reply:
Fair enough.